site/docs/v1/tech/lambdas/index.adoc

```diff
@@ -86,10 +87,33 @@ https://docs.oracle.com/javase/8/docs/technotes/guides/scripting/nashorn/
 
 == Limitations
 
+If the Identity Provider linking strategy is set to `Link Anonymously`, no lambdas will be used by FusionAuth. More information about the link:/docs/v1/tech/identity-providers/#linking-strategies[Identity Provider linking strategies is available here].
+
 The FusionAuth lambdas do not have full access to JavaScript modules and libraries. They also cannot import, require or load other libraries currently. These features might be added to our lambda support in the future but Nashorn does not support them currently.
 
-Please note, if the linking strategy is set to `Link Anonymously`, no lambdas will be used by FusionAuth.
+`console.log` and other `console` methods only take one argument; this differs from the `console` method available in web browsers.
+
+Only the populate or reconcile function is allowed on the top level of a lambda. If you want to have additional functions, create them as function objects inside the main function. For example:
+
+[source,javascript]
+----
+function populate(jwt, user, registration) {
+  var object2String = function(obj) {
+    if (typeof obj === 'number' || typeof obj === 'bigint' || typeof obj === 'boolean') return ('' + obj);
+    if (typeof obj === 'string') return ('"' + obj.replace('"', '\\"') + '"');
+    if (typeof obj === 'function') return ('[function]');
+    if (obj === null) return 'null';
+    if (obj === undefined) return 'undefined';
+    if (Array.isArray(obj)) return '[' + obj.map(function(elem) { return  object2String(elem); }).join(', ') + ']';
+    return '{ ' + Object.keys(obj).map(function (key) { return '"' + key + '": ' + object2String(obj[key]); }).join(', ') + ' }';
+  };
+  console.info('Lambda called');
+  console.info('jwt:' + object2String(jwt));
+----
+
+
+== Future Engines
 
-== Future engines
+The Nashorn engine is being phased out of Java in favor of more robust and advanced engines. 
 
-The Nashorn engine is being phased out of Java in favor of more robust and advanced engines. At any point in the future, we might switch from Nashorn to another JavaScript engine like V8. Therefore, use Nashorn features at your own risk.
+At any point in the future, we might switch from Nashorn to another JavaScript engine such as V8. Therefore, use Nashorn features at your own risk. Please https://github.com/FusionAuth/fusionauth-issues/issues/571[review this GitHub issue for more details].
```

Comment:
What does this mean exactly, can the user even use Nashorn features? Or do we just not know, so we are stating as a CYA?

Reply:
Brian wrote that 3 years ago: https://github.com/FusionAuth/fusionauth-site/blame/63aae4c24cfe91664a82af6d4de986baf1ecd777/site/docs/v1/tech/lambdas/index.adoc

Not sure what that means, other than maybe there are JS engine implementation details that are Nashorn specific?